ops/testing.py

```diff
@@ -2321,6 +2447,53 @@ def _check_protocol_and_port(self, protocol: str, port: Optional[int]):
             raise model.ModelError(f'ERROR invalid protocol "{protocol}", expected "tcp", "udp", or "icmp"\n')  # NOQA: test_quote_backslashes
 
 
+@_copy_docstrings(pebble.ExecProcess)
+class _TestingExecProcess:
+    def __init__(self,
+                 command: List[str],
+                 timeout: Optional[float],
+                 exit_code: Optional[int],
+                 stdin: Union[TextIO, BinaryIO, None],
+                 stdout: Union[TextIO, BinaryIO, None],
+                 stderr: Union[TextIO, BinaryIO, None],
+                 is_timeout: bool):
+        self._command = command
+        self._timeout = timeout
+        self._is_timeout = is_timeout
+        if exit_code is None and not is_timeout:
+            raise ValueError("when is_timeout is False, exit_code must not be None")
+        self._exit_code = exit_code
+        self.stdin = stdin
+        self.stdout = stdout
+        self.stderr = stderr
+
+    def wait(self):
+        if self._is_timeout:
+            raise pebble.TimeoutError(
+                f'timed out waiting for change 123 ({self._timeout} seconds)'
```

Comment:
What I was suggesting here is that we simply remove the change ID, the " 123" part, from the error message entirely. I think it will be confusing to have a hard-coded value here.

Reply:
Gotcha, removed, thanks!